src/software/backend/simulation/physics/physics_field.h

```diff
@@ -54,27 +54,55 @@ class PhysicsField
     Field getFieldWithTimestamp(const Timestamp& timestamp) const;
 
    private:
-    void setupFieldBoundary(std::shared_ptr<b2World> world, const Field& field);
-    void setupEnemyGoal(std::shared_ptr<b2World> world, const Field& field);
-    void setupFriendlyGoal(std::shared_ptr<b2World> world, const Field& field);
+    /**
+     * A helper function that creates a physics body to represent the Field
+     * in the physics world
+     *
+     * @param world The physics world to setup the Field in
+     */
+    void createFieldBody(std::shared_ptr<b2World> world);
+
+    /**
+     * A helper function that creates and sets up physics objects for the field
+     * boundary in the physics world
+     *
+     * @param field The Field being created in the physics world
+     */
+    void setupFieldBoundary(const Field& field);
+
+    /**
+     * A helper function that creates and sets up physics objects for the enemy
+     * goal in the physics world
+     *
+     * @param field The Field being created in the physics world
+     */
+    void setupEnemyGoal(const Field& field);
+
+    /**
+     * A helper function that creates and sets up physics objects for the friendly
+     * goal in the physics world
+     *
+     * @param field The Field being created in the physics world
+     */
+    void setupFriendlyGoal(const Field& field);
+
+    // This body represents the entire field object. Different fixtures and shapes
+    // are added to this body for the field boundary, goals, etc. The fixtures are what
+    // actually handle collisions, which is why we only need a single body to represent
+    // the whole field
+    b2BodyDef field_body_def;
+    b2Body* field_body;
 
-    // This body represents the 4 boundary walls around the edge of the field
-    b2BodyDef field_boundary_body_def;
     b2ChainShape field_boundary_shape;
-    b2Body* field_boundary_body;
     b2FixtureDef field_boundary_fixture_def;
 
-    // This body represents the enemy net
-    b2BodyDef enemy_goal_body_def;
     b2ChainShape enemy_goal_shape;
-    b2Body* enemy_goal_body;
     b2FixtureDef enemy_goal_fixture_def;
 
-    // This body represents the friendly goal
-    b2BodyDef friendly_goal_body_def;
     b2ChainShape friendly_goal_shape;
-    b2Body* friendly_goal_body;
     b2FixtureDef friendly_goal_fixture_def;
 
+    // Since the field never changes during simulation, we store the initial field
+    // used during construction to make it easy to return Field objects when requested
```

Comment:
Bit confused on what the benefit of this is?

Reply:
We can just return the member variable in `getFieldWithTimestamp` rather than recreating a field from the physics objects again.

Reply:
Is there a non-trivial performance benefit that justifies storing duplicate information here?

Reply:
Non-trivial complexity benefit. Reconstructing the `field` from just the physics objects in this case is actually very complicated because of the shapes used, and even then additional information may need to be stored.

I can add that to the comment to help justify.

Reply:
Is it complex enough that it would save a significant amount of time, given that this would only happen once per tick?

Reply:
I more meant _code_ complexity. I think the code would end up being fairly complex.

Reply:
Oh, all good then. Marking as resolved.